plugin/src/testFixtures/kotlin/io/qalipsis/gradle/SSETestServer.kt

```diff
@@ -0,0 +1,126 @@
+package io.qalipsis.gradle
+
+import io.ktor.http.*
+import io.ktor.server.application.install
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.netty.Netty
+import io.ktor.server.response.respondTextWriter
+import io.ktor.server.response.respondText
+import io.ktor.server.response.cacheControl
+import io.ktor.server.routing.post
+import io.ktor.server.routing.get
+import io.ktor.server.routing.RoutingCall
+import io.ktor.server.routing.routing
+import io.ktor.server.sse.SSE
+import kotlinx.coroutines.delay
+import java.time.Instant
+
+/**
+ * Mini server to send server side events over HTTP to facilitate testing.
+ *
+ * @author Francisca Eze
+ */
+object SSETestServer {
+
+    private val server = embeddedServer(Netty, port = 8888) {
+        install(SSE)
+        routing {
+            get("/scenarios") {
+                call.respondText(
+                    """
+                        [
+                          {
+                            "name": "scenario-1",
+                            "description": "First dummy scenario",
+                            "version": "0.0.1"
+                          },
+                          {
+                            "name": "scenario-2",
+                            "description": "Second dummy scenario",
+                            "version": "0.0.1"
+                          }
+                        ]
+                    """.trimIndent()
+                )
+            }
+
+            post("/scenarios") {
+                val query = call.request.queryParameters["type"]
+                when (query) {
+                    "server-shutdown" -> call.sendSuccessAndShutdown()
+                    "401-unauthorized" -> call.unauthorized()
+                    "403-forbidden" -> call.forbidden()
+                    "502-badGateway" -> call.badGateway()
+                    "504-serviceUnavailable" -> call.gatewayTimeout()
+                    "500-internalServerError" -> call.internalServerError()
+                    else -> call.processSuccessResponse()
+                }
+            }
+
+            post("/scenarios") {
+                call.sendSuccessAndShutdown()
+            }
+        }
+    }
+
+    fun start() = server.start(wait = false)
```

Comment:
Documentation on the functions would help to understand the different use cases it covers.

Reply:
Please document all the functions RoutingCall.*